src/sql/query_graph.rs

```diff
@@ -653,20 +700,23 @@ pub fn to_query_graph(st: &SelectStatement) -> Result<QueryGraph, String> {
                     value: l.clone(),
                 }));
             }
-            FieldExpression::Col(ref c) => {
-                match c.function {
-                    None => (), // we've already dealt with this column as part of some relation
-                    Some(_) => {
-                        // add a special node representing the computed columns; if it already
-                        // exists, add another computed column to it
-                        let n = qg.relations
-                            .entry(String::from("computed_columns"))
-                            .or_insert_with(
-                                || new_node(String::from("computed_columns"), vec![], st),
-                            );
-                        n.columns.push(c.clone());
-                    }
+            FieldExpression::Arithmetic(ref a) => {
+                if let ArithmeticBase::Column(ref c) = a.left {
+                    add_computed_column(&mut qg, c);
+                }
```

Comment:
The intention (and if I read this right, the implementation) here is that computed columns (e.g., `SUM(col)`) can occur *within* arithmetic expressions and that we still need to generate the right query graphs for these, correct?

(As opposed to each column defined via an arithmetic expression being considered "computed", which it wouldn't really be from the query graph perspective.)

Reply:
Exactly - before I added this it wasn't possible to refer to computed columns inside arithmetic expressions.

Reply:
:+1: